Sources/_StringProcessing/RegexDSL/Match.swift

```diff
@@ -11,8 +11,24 @@
 
 @dynamicMemberLookup
 public struct RegexMatch<Match> {
+  let input: String
```

Comment:
IIUC, this will (eventually) allow our storage form to be purely range based.

Reply:
This is now purely range based.